tpp-app/src/home/components/Selector.js

```diff
@@ -6,33 +6,75 @@ import MoodIcon from "../../../ios/tppapp/Images.xcassets/icons/mood.svg";
-                <ExerciseIcon style={styles.icon} fill="black"/>
-            </TouchableOpacity>
-            <TouchableOpacity onPress={() => props.toggleSelectedView(props.views.Cramps)} style={[props.selectedView === props.views.Cramps && styles.selectedIcon, styles.iconContainer]}>
-                <CrampsIcon style={styles.icon} fill="black"/>
-            </TouchableOpacity>
-            <TouchableOpacity onPress={() => props.toggleSelectedView(props.views.Sleep)} style={[props.selectedView === props.views.Sleep && styles.selectedIcon, styles.iconContainer]}>
-                <SleepIcon style={styles.icon} fill="black"/>
-            </TouchableOpacity>
+    let flowSelected = props.selectedView === VIEWS.Flow;
+    let moodSelected = props.selectedView === VIEWS.Mood;
+    let exerciseSelected = props.selectedView === VIEWS.Exercise;
+    let crampsSelected = props.selectedView === VIEWS.Cramps;
+    let sleepSelected = props.selectedView === VIEWS.Sleep;
 
+    let selectedColor = "#B31F20";
+    let unselectedColor = "#6D6E71";
+
+
+    return (
+        <View style={[{backgroundColor: '#FFFFFF'}, props.expanded && styles.elevatedSelector]}>
+            {props.expanded &&
+            <View style={[styles.selectorContainer]}>
+                <TouchableOpacity onPress={() => props.toggleSelectedView(VIEWS.Flow)} 
+                    style={[flowSelected && styles.selectedIcon, styles.iconContainer]} >
+                    <FlowIcon style={styles.icon} fill={flowSelected ? selectedColor : unselectedColor}/>
+                </TouchableOpacity>
+                <TouchableOpacity onPress={() => props.toggleSelectedView(VIEWS.Mood)} 
+                    style={[ moodSelected && styles.selectedIcon, styles.iconContainer]}>
+                    <MoodIcon style={styles.icon} fill= {moodSelected ? selectedColor : unselectedColor}/>
+                </TouchableOpacity>
+                <TouchableOpacity onPress={() => props.toggleSelectedView(VIEWS.Exercise)} 
+                    style={[exerciseSelected && styles.selectedIcon, styles.iconContainer]}>
+                    <ExerciseIcon style={styles.icon} fill= {exerciseSelected ? selectedColor : unselectedColor}/>
+                </TouchableOpacity>
+                <TouchableOpacity onPress={() => props.toggleSelectedView(VIEWS.Cramps)} 
+                    style={[ crampsSelected && styles.selectedIcon, styles.iconContainer]}>
+                    <CrampsIcon style={styles.icon} fill= {crampsSelected ? selectedColor : unselectedColor}/>
+                </TouchableOpacity>
+                <TouchableOpacity onPress={() => props.toggleSelectedView(VIEWS.Sleep)} 
+                    style={[sleepSelected && styles.selectedIcon, styles.iconContainer]}>
+                    <SleepIcon style={styles.icon} fill= {sleepSelected ? selectedColor : unselectedColor}/>
+                </TouchableOpacity>
```

Comment:
Is it possible to make this more dynamic? In other words, use `map`

Reply:
Oh yeah good idea. Just made switch.